### PR TITLE
add examples for complex slash commands

### PR DESCRIPTION
**Pull request by lucas-codaio**
Add examples for slash commands that need more context
* Authentication 
* DynamicSyncTable
* Formula
* SyncTable

### CODE REVIEW

File: documentation/.eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    commonjs: true,
+    es6: true,
+    node: true,
+    mocha: true,
+    jest: true,
+  },
+  parser: '@typescript-eslint/parser',
+  plugins: ['ban', 'filenames', 'local', '@typescript-eslint'],
+  settings: {},
+  rules: {
```
Comment (jonathan-codaio) [inline documentation/.eslintrc.js:14]: I'm not an expert in this, but I think eslint merges configuration files together up to the root, so I don't think you have specify anything at all in this file other than the new or modified rules.

File: documentation/.eslintrc.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    '@typescript-eslint/restrict-plus-operands': 'error',
+    'object-shorthand': 'off',
+    '@typescript-eslint/no-unused-vars': [
+      'error',
+      {varsIgnorePattern: 'response', argsIgnorePattern: 'myParam|context'},
```
Comment (jonathan-codaio) [inline documentation/.eslintrc.js:19]: Nice

File: documentation/documentation_config.ts [hunk 3]
```diff
@@ -149,4 +149,27 @@ export const Examples: Example[] = [
     codeFiles: ['./examples/column-format/no_matchers.ts'],
     categories: [Category.ColumnFormat],
   },
+  {
+    contentFile: './examples/authentication/authentication.md',
+    codeFiles: [
+      './examples/authentication/system_authentication.ts',
+      './examples/authentication/user_authentication.ts',
+    ],
+    categories: [Category.Authentication],
```
Comment (jonathan-codaio) [inline documentation/documentation_config.ts:158]: I forgot to mention this before, but we need to specify trigger words here too, right? I.e. the word(s) that you can hover over to be prompted to open this example in the side panel.

File: documentation/examples/authentication/system_authentication.ts [hunk 4]
```diff
@@ -0,0 +1,11 @@
+import * as coda from '../../../index';
+
+const pack = coda.newPack();
+
+// BEGIN
+
+pack.setSystemAuthentication({
+  // Replace HeaderBearerToken with an authentication type
+  // besides OAuth2, CodaApiHeaderBearerToken, None and Various.
```
Comment (jonathan-codaio) [inline documentation/examples/authentication/system_authentication.ts:9]: Nit, would simplify this second line to something like "(Not all authentication types are available for system authentication.)"

File: documentation/examples/authentication/user_authentication.ts [hunk 5]
```diff
@@ -0,0 +1,10 @@
+import * as coda from '../../../index';
+
+const pack = coda.newPack();
+
+// BEGIN
+
+pack.setUserAuthentication({
+  // Replace None with the authentication type that applies for your connection.
```
Comment (jonathan-codaio) [inline documentation/examples/authentication/user_authentication.ts:8]: Nit, maybe `for your pack.`

File: documentation/examples/dynamic-sync-table/basic_dynamic_sync_table.ts [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+    return response.body.browserLink;
+  },
+  formula: {
+    // This is the name that will be called in the formula builder.
+    // Remember, your formula name cannot have spaces in it.
+    name: '<DynamicSyncTable>',
+    description: '<Creates a dynamic sync table>',
+
+    // If your formula requires one or more inputs, you’ll define them here.
+    // You can change the coda.Type to anything but object.
```
Comment (jonathan-codaio) [inline documentation/examples/dynamic-sync-table/basic_dynamic_sync_table.ts:58]: Nit, this is obsolete now, can remove.

**Review by jonathan-codaio — COMMENTED** [review]
Cool

File: documentation/documentation_config.ts (hunk 3 above)
Comment (lucas-codaio) [reply documentation/documentation_config.ts:158]: Ran into this issue when connecting 😆

File: documentation/documentation_compiler.ts [hunk 7]
```diff
@@ -37,6 +37,7 @@ function compileExamples() {
       content,
       code,
       categories: example.categories,
+      triggerWords: example.triggerWords,
```
Comment (jonathan-codaio) [inline documentation/documentation_compiler.ts:40]: My bad, maybe let's call these `triggerTokens` to be precise.

**Review by jonathan-codaio — APPROVED** [review]

**Pull request merged**